protos/public_api.proto

```diff
@@ -160,11 +169,32 @@ message LoginResponse {
     string jwt_token = 2;
 }
 
-message ResetPasswordRequest {
+// Used for Signing Up a new User and Resetting a Password
+message SetPasswordRequest {
     string password = 1;
+}
+
+message ChangePasswordRequest {
+    string old_password = 1;
     string new_password = 2;
 }
 
+enum InviteAction {
```

Comment:
we should not expose this to APIs, instead it is understood or implicit by the RPC name (SetPassword, ChangePassword)

Reply:
SetPassword is different from Change Password. Set Password is for Signup(Add new user) and Reset Password.

Reply:
Invitations for both `UserAddInvite` and `UserResetPasswordInvite` in their corresponding handlers